packages/pass-style/src/passStyleOf.js

```diff
@@ -63,6 +68,21 @@ const makeHelperTable = passStyleHelpers => {
   return harden(HelperTable);
 };
 
+/**
+ * The `assertRestValid` assumes that the `canBeValid` check has already passed.
+ * Contexts where we cannot assume that should call `assertValid` instead,
+ * which checks both conditions in the right order.
+ *
+ * @param {PassStyleHelper} helper
+ * @param {any} candidate
+ * @param {(val: any) => PassStyle} passStyleOfRecur
+ * @returns {void}
+ */
+const assertValid = (helper, candidate, passStyleOfRecur) => {
+  helper.canBeValid(candidate, assertChecker);
+  helper.assertRestValid(candidate, passStyleOfRecur);
```

Comment:
I really like the switch to this pattern, but I wonder if we can come up with a better name than `assertRestValid`. Maybe `assertActuallyValid`? or `assertCompleteValidity`?

Reply:
I am open to a new term. But the overriding issue is a name that indicates the it isn't providing assurance by itself, but only if it follows another check. Both `assertActuallyValid` and `assertCompleteValidity` are likely to be misread this way. The second misreading is amusing: "complete" as a verb or an adjective?

Reply:
That's true. Most accurate would be something like `assertConditionallyValid`, which is unfortunately a mouthful. We're really doing material condition in an unusual decomposition, where `assertRestValid` is "A ⇒ B" and `canBeValid` is "A", such that the combination in temporal order is "A ∧ (A ⇒ B)" = "B".

Reply:
Good. Resolving for now. But unresolve if you have a new idea. Thx.